memcache/memcache_test.go

```diff
@@ -261,6 +264,87 @@ func testTouchWithClient(t *testing.T, c *Client) {
 	}
 }
 
+func testMetaCommandsWithClient(t *testing.T, c *Client, checkErr func(err error, format string, args ...interface{})) {
+	// Meta Set
+	key := "bah"
+	opaqueToken := "A123"
+	metaFoo := &MetaSetItem{Key: key, Value: []byte("bahval"), Flags: MetaSetFlags{ReturnKeyInResponse: true, ReturnCasTokenInResponse: true, OpaqueToken: &opaqueToken}}
+	response, err := c.MetaSet(metaFoo)
+	if *response.Key != key {
+		t.Errorf("meta set(%s) Key = %q, want %s", key, *response.Key, key)
+	}
+	if *response.OpaqueValue != opaqueToken {
+		t.Errorf("meta set(%s) Opaque token = %s, want %s", key, *response.Key, opaqueToken)
+	}
+	casToken := response.CasId
+	if casToken == nil {
+		t.Errorf("meta set(%s) error, no CAS token returned", key)
+	}
+	checkErr(err, "first meta set(%s): %v", key, err)
+
+	// set using the same cas token as what was last set
+	var newTTL int32 = 900000
+	var clientFlagToken uint32 = 90
+	metaFoo = &MetaSetItem{Key: key, Value: []byte("new_bah_val"), Flags: MetaSetFlags{CompareCasTokenToUpdateValue: casToken, ClientFlagToken: &clientFlagToken, UpdateTTLToken: &newTTL}}
+	response, err = c.MetaSet(metaFoo)
+	checkErr(err, "second meta set(%s): %v", key, err)
+
+	// set with no reply semantics turned on
+	// note that the documentation says that this flag will always return an error (even if the command runs successfully)
+	metaFoo = &MetaSetItem{Key: key, Value: []byte("with_base64_key"), Flags: MetaSetFlags{UseNoReplySemanticsForResponse: true}}
+	_, err = c.MetaSet(metaFoo)
+	if err == nil {
```

Comment:
Yea you don't see an error on local memcached but the API will always return an error, as per the [documentation](https://github.com/memcached/memcached/blob/046c4bb5d8498420c13e5357c8299b60952b2595/doc/protocol.txt#L608).

Reply:
Yes. I read the documentation but I believe what they are saying is it will always return an error "if" error occurs. I just tried a simple meta set on a non existing key and later did a set with an invalid cas token on the same key and I got error on the second command not first. I believe you are getting an error because your code is hitting the last return in your processMetaSet function. I believe we should only return error if an error actually occurs. Let me know if my understanding is correct.